pkg/planner/core/scalar_subq_expression.go

```diff
@@ -178,9 +178,9 @@ func (*ScalarSubQueryExpr) IsCorrelated() bool {
 	return false
 }
 
-// ConstItem implements the Expression interface.
-func (*ScalarSubQueryExpr) ConstItem(_ bool) bool {
-	return true
+// ConstLevel returns the const level for the expression
+func (*ScalarSubQueryExpr) ConstLevel() expression.ConstLevel {
+	return expression.ConstNone
```

Comment:
Looks like the original logic is return ConstStrict here?

Reply:
Yes， confirmed with @qw4990 , returning `ConstNone` instead `ConstStrict` should be more safe